### ISSUE TITLE
Filter out closed AdSense accounts based on new API capabilities

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#5050]
Based on the recently released new AdSense API capabilities and the [corresponding enhancements to the design doc](https://docs.google.com/document/d/1vNpMW4iED2scWOjpF8SKGoM8z6GUjX-WbhMpmAKeieo/edit#heading=h.dvwukljp9z7x), we need to ensure closed AdSense accounts are ignored by Site Kit. For example, if a user has only a single closed account, they will need to create a new AdSense account, which is the same as if they had no account at all.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The AdSense `GET:accounts` data point should filter out any accounts from the API response that have a `state: CLOSED`, as they can be ignored from our perspective.
    * See [the new account state documentation](https://developers.google.com/adsense/management/reference/rest/v2/accounts#State) for background on the possible values.

## Implementation Brief
* Using `includes/Modules/AdSense.php`,
	* Add a new public static function `is_account_not_closed ` which should have the `account` array as parameter. The function should return a boolean, whether `account.state` is different from `CLOSED`.
	* Within `GET:accounts` in `parse_data_response`, instead of returning the result of `$response->getAccounts` (with the filtered account IDs), store the accounts in a variable (`$accounts`) and then use `array_filter` to filter the array, passing the array (`$accounts`) as first parameter and the function `filter_account_with_state_closed` as second parameter. Return the updated array.

### Test Coverage
* No new tests to be added.

## QA Brief

* Use a Google account where you have a closed AdSense account.
* Start the AdSense module setup and verify that this account does not show up or that there are no errors due to this account being chosen.
* Basically, if that closed AdSense account is the only one you have, you should land in the UI to create a new AdSense account.
* This should be tested both with and without the `adsenseSetupV2` flag enabled.
* This isn't exactly possible to set up in a reasonable timeframe, but @felixarntz has such an account, so he is going to QA this issue.

## Changelog entry

* Ensure closed AdSense accounts are not considered for the AdSense account to use with the module.

**Comment by sancodes** [comment]
If this is open I would like to help. Please let me know

**Comment by felixarntz** [comment]
@asvinb Just one thing regarding the IB: The `filter_account_with_state_closed` doesn't really sound accurate. Since it's a boolean check for whether the account is _not_ closed, how about we call it `is_account_open` or `is_account_not_closed` or something like that? Other than that the IB looks good.

**Comment by felixarntz** [comment]
@sancodes Absolutely, your contribution would be welcome! We need to complete this issue within the next week to get it into the upcoming Site Kit release, so if you could work on a pull request in the next few days that would be great. Before starting the pull request, please review the "Implementation Brief" in the issue description. It describes the technical approach to take, so you can follow that for the PR. It will probably also help you especially as you're new to the PHP codebase.

**Comment by sancodes** [comment]
@felixarntz will do thanks.

**Comment by asvinb** [comment]
@felixarntz IB has been updated as per your feedback. Can you take another look? Thanks!

**Comment by sancodes** [comment]
@felixarntz will start on it thank you.

**Comment by felixarntz** [comment]
## QA ✅ 

* Before this fix, starting the AdSense setup flow causes an error because it chooses my closed AdSense account.
    * The account is returned by our API endpoint.
    * The UI keeps loading due to the unexpected error.
* After this fix, starting the AdSense setup flow works correctly since my closed AdSense account is now ignored.
    * The account is no longer returned by our API endpoint.
    * The UI now shows me the "Create account" step (since I have no other account than the closed one).
* This behaved the same regardless of the `adsenseSetupV2` flag.

### Screenshots
**Before**
<img width="1581" alt="Screen Shot 2022-04-15 at 10 46 52 AM" src="https://user-images.githubusercontent.com/3531426/163604065-7b0e70b6-174e-4272-b7bd-50a8ff00f1e8.png">

**After**
<img width="1625" alt="Screen Shot 2022-04-15 at 10 47 15 AM" src="https://user-images.githubusercontent.com/3531426/163604090-9ab3c00e-1ff2-4604-9c8d-0a19303e68a2.png">

**Comment by wpdarren** [comment]
@felixarntz I am creating the test plan for AdSense Setup Improvement and wondering if this could now be tested by us since we have the tester plugin update. What are your thoughts? If it can, would it be possible for you to write up a new QAB, and I can make sure that this gets included. 🤔

**Comment by felixarntz** [comment]
@wpdarren Since this is something which happens on the server-side, it wouldn't really be possible to mock this via the tester plugin.

I think this piece is simple enough so that we don't need to cover this as part of the bug bash.
